From 8d2a6325a891a2fab8bf7c091cb8fb73468ae1ea Mon Sep 17 00:00:00 2001
From: Dmitry Timoshkov <dmitry@baikal.ru>
Date: Thu, 20 Oct 2016 16:57:51 +0800
Subject: windowscodecs: WICConvertBitmapSource should ask
 IWICFormatConverter::Initialize to use an optimized palette.

This matches Windows' behaviour, and makes a simple test application that converts
any WIC compatible RGB(A) source image to 8bppIndexed format (GIF for instance) work.
---
 dlls/windowscodecs/info.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/windowscodecs/info.c b/dlls/windowscodecs/info.c
index 1a93491..c1e8a66 100644
--- a/dlls/windowscodecs/info.c
+++ b/dlls/windowscodecs/info.c
@@ -2336,7 +2336,7 @@ HRESULT WINAPI WICConvertBitmapSource(REFWICPixelFormatGUID dstFormat, IWICBitma
 
                     if (SUCCEEDED(res) && canconvert)
                         res = IWICFormatConverter_Initialize(converter, pISrc, dstFormat, WICBitmapDitherTypeNone,
-                            NULL, 0.0, WICBitmapPaletteTypeCustom);
+                            NULL, 0.0, WICBitmapPaletteTypeMedianCut);
 
                     if (FAILED(res) || !canconvert)
                     {
-- 
2.9.0

